src/accelerate/commands/launch.py

```diff
@@ -350,6 +397,7 @@ def deepspeed_launcher(args):
         warnings.warn('--fp16 flag is deprecated. Use "--mixed_precision fp16" instead.', DeprecationWarning)
         mixed_precision = "fp16"
 
+    current_env["PYTHONPATH"] = sys.executable
```

Comment:
this is a problem, @pacman100 - how can a user then use `PYTHONPATH` if it's being hijacked by the launcher?

If you must manipulate `PYTHONPATH` you need to extend it and not override it.

Reply:
but also why would you set something that should be a path to find python libraries to a path to an executable?

Reply:
Proposed fix https://github.com/huggingface/accelerate/pull/741

Reply:
I don't remember but it was needed to know the python env on other nodes to run the code. If I don't do that, deepspeed code will still hijack it: https://github.com/microsoft/DeepSpeed/blob/master/deepspeed/launcher/runner.py#L480-L484.

Below is the run with this line commented and you can notice that PYTHONPATH is still overridden by the above deepspeed code:

<img width="1502" alt="Screenshot 2022-10-05 at 12 35 26 PM" src="https://user-images.githubusercontent.com/13534540/194000694-8d5aa407-7ff3-42a0-9f84-908e8ae67713.png">

Reply:
1. it doesn't hijack the setting, it appends to it:

```
        curr_path = os.path.abspath('.')
        if 'PYTHONPATH' in env:
            env['PYTHONPATH'] = curr_path + ":" + env['PYTHONPATH']
        else:
            env['PYTHONPATH'] = curr_path
```

2. it doesn't add `sys.executable` it adds a directory.

Reply:
Hello Stas, yeah, you are correct, my bad. After spending quite some time diving deep and setting up the multi-node setup to test things out, I have found the reason.

In Accelerate, all the DeepSpeed config from `accelerate config` command is passed as env variables, as shown towards the end of the above screenshot. DeepSpeed uses `.deepspeed_env` file for all these 
additional env variables instead of using the passed env variables as mentioned here: https://www.deepspeed.ai/getting-started/#multi-node-environment-variables

So, I was adding all the env variables to `.deepspeed_env` file as seen here: https://github.com/huggingface/accelerate/blob/main/src/accelerate/commands/launch.py#L590-L594
. The DeepSpeed snippet from the above message would extend/write to PYTHONPATH. However, just after that things will be read from `.deepspeed_env` and env variables will be reset or newly added. Here, the `pwd` extension would be reset and the original PYTHONPATH if sent from accelerate will be used instead of extended one. I overlooked extension usecase and confused `pwd` for python env path and just added that directly to the launch env variables leading to unintended behaviour.

Reply:
Thank you for the extended commentary that gives more context to the problem, @pacman100 

And it looks like your suggestion in the PR resolving this issue. which has been merged now, so resolving this conversation.